File: piper/android/app/src/main/java/io/gopiper/piper/engine/statements/math_modulo.kt
package io.gopiper.piper.engine.statements

import android.util.Log
import io.gopiper.piper.engine.Piper
import org.json.JSONObject
import java.lang.Exception

suspend fun Piper.executeMathModulo(statement: JSONObject, onError: (e: Exception) -> Unit): Any? {
    try {
        Log.d("hello", "execute_math_modulo")
        val pointer = statement.getJSONObject("pointer")
        val value1 = executeStatement(pointer["value1"].toString())
        val value2 = executeStatement(pointer["value2"].toString())

        val out = value1.toString().toDouble() % value2.toString().toDouble()
        return if (out.toString().endsWith(".0")) {
            out.toInt()
        } else {
            out
        }
    } catch (e: Exception) {
        onError(e)
    }
    return null
}

File: piper/extensions/chrome/src/engine/statements/math_modulo.js
import { stringify } from "../util"


export default async function(statement, onError) {
    try {
        let pointer = statement.pointer
        let value1 = await this.executeStatement(stringify(pointer.value1))
        let value2 = await this.executeStatement(stringify(pointer.value2))

        return value1 % value2
    } catch (e) {
        onError(e)
    }

    return null
}
